src/Utils/normalization.jl

```diff
@@ -69,31 +61,72 @@ function normalize(system::AbstractSystem)
     throw(ArgumentError("the system type $(typeof(system)) is currently not supported"))
 end
 
-# systems already in canonical form, i.e. which don't need normalization
-normalize(system::CF) = system
+# x' = Ax, in the continuous case
+# x+ = Ax, in the discrete case
+function normalize(system::LCS{N, AN}) where {N, AN<:AbstractMatrix{N}}
+    n = statedim(system)
+    X = Universe(n)
+    return CLCS(system.A, X)
+end
+
+function normalize(system::LDS{N, AN}) where {N, AN<:AbstractMatrix{N}}
+    n = statedim(system)
+    X = Universe(n)
+    return CLDS(system.A, X)
+end
+
+# x' = Ax, x ∈ X in the continuous case
+# x+ = Ax, x ∈ X in the discrete case
+for CL_S in (:CLCS, :CLDS)
+    @eval begin
+        function normalize(system::$CL_S{N, AN, ST}) where {N, AN<:AbstractMatrix{N}, ST<:LazySet{N}}
+            n = statedim(system)
+            X = _wrap_invariant(stateset(system), n)
```

Comment:
I think you want to remove the type constraints. Otherwise you cannot pass `nothing` as invariant and so the call to `_wrap_invariant` is useless. In case you decided to forbid `nothing` as invariant (with which I agree here - just use a different system type), the call to `_wrap_invariant` should be removed.
(The same applies to the method below.)

Reply:
Thanks -- i should have used `XT<:XNCF` here.

Reply:
Then you need to rename `ST` to `XT`.

Reply:
A reason to keep `nothing` is that it comes natural in Julia, as means to say "there is no value". The other reason is to handle generic macro notation such as

```julia
@system x' = A*x + B*u, u in U
```
without specifying the state constraints, i would tend to let `X` be "nothing" and wrap it as a `CLCCS`.